pkgs/development/python-modules/mahotas/default.nix

```diff
@@ -1,24 +1,30 @@
-{ buildPythonPackage, fetchFromGitHub, nose, pillow, scipy, numpy, imread, lib, stdenv }:
+{ buildPythonPackage, fetchFromGitHub, pillow, scipy, numpy, pytest, imread, freeimage, lib, stdenv }:
 
 buildPythonPackage rec {
   pname = "mahotas";
-  version = "1.4.10";
+  version = "1.4.11";
 
   src = fetchFromGitHub {
     owner = "luispedro";
     repo = "mahotas";
     rev = "v${version}";
-    sha256 = "0fjiyl82wj1a6xzr9mss2y2rydl4zchl2cbdbg0jm0fcrs99q4hw";
+    sha256 = "029gvy1fb855pvxvy8zwj44k4s7qpqi0161bg5wldfiprrysn1kw";
   };
 
   # remove this as soon as https://github.com/luispedro/mahotas/issues/97 is fixed
   patches = [ ./disable-impure-tests.patch ];
 
-  propagatedBuildInputs = [ numpy imread pillow scipy ];
-  checkInputs = [ nose ];
+  propagatedBuildInputs = [ numpy imread pillow scipy freeimage ];
+  checkInputs = [ pytest ];
 
-  checkPhase= ''
-    python setup.py test
+  postPatch = ''
+    substituteInPlace mahotas/io/freeimage.py --replace "/opt/local/lib" "${freeimage}/lib"
+  '';
+
+  # tests must be run in the build directory
+  checkPhase = ''
+    cd build/lib*
+    pytest
   '';
 
   disabled = stdenv.isi686; # Failing tests
```

Comment:
Do they still fail?

Reply:
I didn't check and rather would leave that to someone having the machinery in place.